packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -235,6 +236,14 @@ export abstract class BaseFetchService<
         continue;
       }
 
+      if (cachedDictBlocks.length > 0) {
```

Comment:
I think this is starting to do what this issue is wanting to achieve https://github.com/subquery/subql/issues/1960.

This is effectively like adding another queue which is unnecessary. We should look at changing the behaviour of the block number queue to compensate for this. I also think we should set a limit on the number of dictionary results (100 per entity is the default so lets stick with that)